src/jit/compiler.cpp

```diff
@@ -718,13 +712,25 @@ var_types  Compiler::getArgTypeForStruct(CORINFO_CLASS_HANDLE clsHnd,
         useType = GetEightByteType(structDesc, 0);
     }
 
-#else // not UNIX_AMD64
+#elif defined(_TARGET_X86_)
 
-    // We set the "primitive" useType based upon the structSize
-    // and also examine the clsHnd to see if it is an HFA of count one
-    useType = getPrimitiveTypeForStruct(structSize, clsHnd);
+    // On x86 we pass all structs by value 
+    useType = TYP_UNKNOWN;
```

Comment:
On x86 the VM will unwrap single field structs and then give the JIT a primitive type, so we never see the type as a struct.  On the other platforms/targets the VM tells us it is a TYP_STRUCT and the JIT will examine the size of the struct and choose to pass and return it as a primitive when appropriate.

So I changed the comment to:

  // On x86 we never pass structs as primitive types (unless the VM unwraps them for us)